tests/unit/objects/test_ensure_type_hints.py

```diff
@@ -0,0 +1,84 @@
+"""
+Ensure type-hints are setup correctly and detect if missing functions.
+
+Original notes by John L. Villalovos
+
+"""
+import inspect
+import pprint
+
+import pytest
+
+import gitlab.mixins
+import gitlab.v4.objects
+
+
+def test_check_get_methods():
+    """Ensure classes derives from ListMixin have defined a 'get()' method with
+    correct type-hints.
+    """
+
+    failed_messages = []
+    for module_name, module_value in inspect.getmembers(gitlab.v4.objects):
+        if not inspect.ismodule(module_value):
+            # We only care about the modules
+            continue
+        # Iterate through all the classes in our module
+        for class_name, class_value in inspect.getmembers(module_value):
+            if not inspect.isclass(class_value):
+                continue
+
+            if not class_name.endswith("Manager"):
+                continue
+
+            # Ignore imported classes from gitlab.base
+            if class_value.__module__ == "gitlab.base":
+                continue
+
+            # print("class_name:", class_name)
+            mro = class_value.mro()
+
+            if gitlab.mixins.GetMixin not in mro:
+                # print("mro", mro)
+                continue
+
+            obj_cls = class_value._obj_cls
+            signature = inspect.signature(class_value.get)
+
+            # pprint.pprint(class_value)
+            # print(obj_cls)
+            # print("def get", signature)
+            # print(signature.return_annotation)
+
+            if obj_cls != signature.return_annotation:
+                filename = inspect.getfile(class_value)
+                # print("Bad return annotation")
+                failed_messages.append(
+                    (
+                        f"class definition for {class_name!r} in file {filename!r} "
+                        f"must have defined a 'get' method with a return annotation of "
+                        f"{obj_cls} but found {signature.return_annotation}"
+                    )
+                )
```

Comment:
The way I'd probably do this is have an assert per class, so instead of looping over classes in a single test case I'd provide them as parametrized fixtures or something like that.
Then we don't need to manually construct the error message. But maybe I'm just obsessed with pytest and it's not needed :)

Reply:
But then don't you only show one error and stop? At the moment we have like 100+ to fix. I like to be able to see every one that fails.

If there is a way to show every failure as an assert that would be great!

Reply:
I think pytest should show all errors unless it's passed `--exitfirst`. But if it's too complicated to set up with fixtures maybe it's not worth it :)

Reply:
I think I figured it out!